### PR TITLE
Define method getDefaultValueAnnotatedType

### PR DESCRIPTION
**Pull request by mernst**
The CI failures are due to missing "WARNING: Illegal reflective access" in `expected.txt` files.  I'd like to find a way to avoid those warnings.

They occur because this implementation uses reflection to access private members.  What is a better implementation?

In case it helps, my end goal is to determine whether a field's annotation is compatible with the default value that Java uses for uninitialized fields.  That is, the client code looks like this:

```java
            AnnotatedTypeMirror fieldType = defaultValueAtypeFactory.getAnnotatedType(field);
            AnnotatedTypeMirror defaultValueType =
                    defaultValueAtypeFactory.getDefaultValueAnnotatedType(
                            fieldType.getUnderlyingType());
            if (!defaultValueAtypeFactory
                    .getTypeHierarchy()
                    .isSubtype(defaultValueType, fieldType)) {
```

### CODE REVIEW

File: framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java
Comment (smillst) [inline framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java:2247]: `org.checkerframework.javacutil.trees.TreeBuilder#buildLiteral` seems to build a literal tree from a given value.  

You should be able to do something like this:

```
    public static LiteralTree buildLiteral(Object value, ProcessingEnvironment env) {
        Context context = ((JavacProcessingEnvironment) env).getContext();
         TreeMaker maker = TreeMaker.instance(context);
        return maker.Literal(value);
    }
```

(This sort of method should be in TreeUtils.)

In case it's not easy to figure out.  Here are the imports for the code above:
```
import com.sun.tools.javac.tree.TreeMaker;
import javax.annotation.processing.ProcessingEnvironment;
import com.sun.tools.javac.util.Context;
import com.sun.tools.javac.processing.JavacProcessingEnvironment;
```

**Comment by mernst** [comment]
@smillst This is ready for re-review, but don't merge it yet because I am still running tests on clients of it.

File: javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java
Comment (smillst) [inline javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java:1664]: ` maker.Literal(TypeTag.BOT, null)` should be ` maker.Literal(typeTag, value)`, right?  Why doesn't `maker.Literal(typeTag, value)` work? I don't think there are any test for this, so I can't check myself.

File: framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java
Comment (smillst) [inline framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java:2223]: This should be in `TreeUtils` because it imports a class from outside the support complier API, (com.sun.tools.javac.code.TypeTag).

**Review by smillst — APPROVED** [review]
It's fine with me if you want to merge this, after addressing my last round of comments.

**Comment by mernst** [comment]
Thanks.  That fixed the downstream jobs.

**Pull request merged**
